### PR TITLE
add function to get local address name

### PR DESCRIPTION
**Pull request by jebrando**
add the ability to get local address name from the socket

### CODE REVIEW

File: interfaces/devdoc/socket_transport_requirements.md [hunk 1]
```diff
@@ -171,4 +178,12 @@ MOCKABLE_FUNCTION(, SOCKET_HANDLE, socket_transport_get_underlying_socket, SOCKE
...
\ No newline at end of file
+`socket_transport_check_valid_handle` checks that the internal socket is valid.
+
+### socket_transport_get_local_address
+
+```c
+MOCKABLE_FUNCTION(, int, socket_transport_get_local_address, SOCKET_TRANSPORT_HANDLE, socket_transport, char, hostname[MAX_GET_HOST_NAME_LEN], LOCAL_ADDRESS**, local_address_list, uint32_t*, address_count);
+```
+
+`socket_transport_get_local_address` shall return the hostname and IP address of the machine.
```
Comment (nishudeshu) [inline interfaces/devdoc/socket_transport_requirements.md:189]: local machine.

Comment (nishudeshu) [inline interfaces/devdoc/socket_transport_requirements.md:189]: should it be plural? Does it return more than one address if there are multiple interface cards?

File: linux/devdoc/socket_transport_linux_requirements.md [hunk 2]
```diff
@@ -344,4 +345,34 @@ MOCKABLE_FUNCTION(, bool, socket_transport_is_valid_socket, SOCKET_TRANSPORT_HAN
...
 **SOCKET_TRANSPORT_LINUX_11_094: [** If the socket inside `socket_transport_handle` is an `INVALID_SOCKET`, `socket_transport_is_valid_socket` shall fail and return `false`. **]**
 
-**SOCKET_TRANSPORT_LINUX_11_095: [** On success, `socket_transport_is_valid_socket` shall return `true`. **]**
\ No newline at end of file
+**SOCKET_TRANSPORT_LINUX_11_095: [** On success, `socket_transport_is_valid_socket` shall return `true`. **]**
+
+### socket_transport_get_local_address
+
+```c
+MOCKABLE_FUNCTION(, int, socket_transport_get_local_address, SOCKET_TRANSPORT_HANDLE, socket_transport, char, hostname[MAX_GET_HOST_NAME_LEN], LOCAL_ADDRESS**, local_address_list, uint32_t*, address_count);
```
Comment (nishudeshu) [inline linux/devdoc/socket_transport_linux_requirements.md:353]: >socket_transport_get_local_address [](http://example.com/codeflow?start=25&length=34)

Is there a way we can specify the interface card or default if that field is NULL?

Comment (jebrando) [reply linux/devdoc/socket_transport_linux_requirements.md:353]: I thought about that, but the problem is we don't know what interface card to specify.  We don't have the name of them or the value.

File: interfaces/devdoc/socket_transport_requirements.md (hunk 1 above)
Comment (jebrando) [reply interfaces/devdoc/socket_transport_requirements.md:189]: changed.

Comment (jebrando) [reply interfaces/devdoc/socket_transport_requirements.md:189]: good catch.

File: linux/devdoc/socket_transport_linux_requirements.md (hunk 2 above)
Comment (nishudeshu) [reply linux/devdoc/socket_transport_linux_requirements.md:353]: Okay, usually developers just provide the string field and run some command to get all available NIC cards and compare them to the input value. But not needed for this PR, I guess.

**Review by nishudeshu — APPROVED** [review]

**Pull request merged**